binary_search_tree/tree.py

```diff
@@ -14,46 +14,136 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: 0(log(n)
+    # Space Complexity: 0(1)
     def add(self, key, value = None):
-        pass
+        added = TreeNode(key, value)
+        if not self.root:
+            self.root = added
```

Comment:
If you do not return after this guard close, you'll enter into an infinite loop
```suggestion
            self.root = added
            return added
```